static/js/components/dashboard/CourseAction.js

```diff
@@ -87,55 +89,83 @@ export default class CourseAction extends React.Component {
     );
   }
 
+  renderDescription(text: string, className: string): React$Element<*> {
+    return text.length > 0 ?
+      <div className={className}>{text}</div> :
+      null;
+  }
+
+  renderTextDescription(text: string): React$Element<*> {
+    return this.renderDescription(text, 'description');
+  }
+
+  renderBoxedDescription(text: string): React$Element<*> {
+    return this.renderDescription(text, 'boxed description');
+  }
```

Comment:
ya know, i tried getting this to work before but i was somehow screwing up the usage of `R.curry`. i love me some FP so thanks for the leg up!